dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/accounts/UserServlet.java

```diff
@@ -40,37 +43,46 @@
 public class UserServlet extends HttpServlet {
 
     @Override
-    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String usernameToRemove = req.getParameter("username");
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        String regex = "/(.+?)/.+";       // matches and groups the word(s) between "/"
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(req.getPathInfo());
+
+        String usernameToRemove = "";
+
+        if (matcher.find()) {
+            usernameToRemove = matcher.group(1);        // assigns first group of the match
+        }
 
         boolean isRemoved = false;
         if (usernameToRemove != null && !usernameToRemove.equals("")) {
             isRemoved = UsersStruct.getInstance().removeUser(usernameToRemove);
         }
 
-        ResponseUtil.simpleResponse(resp, "success",isRemoved);
+        ResponseUtil.simpleResponse(resp, "success", isRemoved);
     }
 
     @Override
-    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
```

Comment:
This is a public web API change. Is it really important that we change the method? Couldn't `PUT` be adapted to also enable updating existing users?

Reply:
Since the doPost assumes the creation of a new element; since doPut assumes the edition/replacement of a new element; since, according to literature, the previous format was not in compliance. I believe this is the best way to do it.
https://spring.io/understanding/REST

Reply:
Again, we can think of this the other way around: if this implementation `PUT` does not support updating a user's properties, that can be improved. As much as we'd like to have a "pure" RESTful interface, we do not mandate such a design in Dicoogle. This part of the API was designed like this, and changing it is something we should not be accustomed to, since it has to be changed in client-sided code.